lib/profile/commands/view.rb

```diff
@@ -11,15 +11,43 @@ class View < Command
       def run
         @name = args[0]
         raise "Node '#{@name}' not found" unless node
+
+        if @options.watch
+          in_clean_window do
+            loop do
+              height = `tput lines`.chomp.to_i
+              print "\r\e[#{height}A"
+              system("clear")
+              puts output.lines.pop(height)
+              sleep(0.5)
+            end
+          end
+        else
+          puts output
+        end
+      end
+
+      def output
         log = File.read(node.log_file)
         commands = log.split(/(?=PROFILE_COMMAND)/)
-        commands.each { |cmd| puts command_structure(cmd) }
+        "".tap do |output|
+          commands.each { |cmd| output << command_structure(cmd) + "\n" }
+        end
+      end
+      
+      def in_clean_window
+        system "tput smcup"
+        begin
+          yield
+        rescue Interrupt
+          system "tput rmcup"
+        end
```

Comment:
This is an excellent idea. Did you come up with this?

Reply:
I wasn't aware of the existence of `tput smcup` until @ColonelPanicks pointed out a helpful stackoverflow post about it, although while I was implementing it for `--watch` it felt more natural for `in_clean_window` to exist, and it may well make for a useful function to be used elsewhere should we ever have a need for something like this again.

Reply:
Understanding `block` / `yield` calls is an important part of developing Ruby, so good job.